token/program-2022/src/extension/mod.rs

```diff
@@ -505,13 +505,19 @@ impl<'data, S: BaseState> StateWithExtensionsMut<'data, S> {
         &self,
         new_extension: ExtensionType,
     ) -> Result<Option<usize>, ProgramError> {
-        let current_extensions = self.get_extension_types()?;
-        let needed_tlv_len = ExtensionType::get_total_tlv_len(&current_extensions);
-        let new_needed_tlv_len = needed_tlv_len.saturating_add(new_extension.get_type_len());
+        let mut extensions = self.get_extension_types()?;
+        if !extensions.contains(&new_extension) {
+            extensions.push(new_extension);
+        }
+        let new_needed_tlv_len = ExtensionType::get_total_tlv_len(&extensions);
         if self.tlv_data.len() >= new_needed_tlv_len {
             Ok(None)
         } else {
-            Ok(Some(new_needed_tlv_len - self.tlv_data.len())) // arithmetic safe because of if clause
+            let mut diff = new_needed_tlv_len - self.tlv_data.len(); // arithmetic safe because of if clause
+            if self.account_type.is_empty() {
+                diff = diff.saturating_add(size_of::<AccountType>());
+            }
```

Comment:
Is this being used on mints? If so, we'll have to also add the padding required.

Reply:
Mm, true. I don't know of an immediate use-case for Mints, aside from my tests, but should support it properly. Thanks!